src/autopas/containers/verletListsCellBased/verletLists/traversals/VLListIterationTraversal.h

```diff
@@ -46,10 +46,26 @@ class VLListIterationTraversal : public TraversalInterface, public VLTraversalIn
   void initTraversal() override {
     auto &cells = *(this->_cells);
     if (dataLayout == DataLayoutOption::soa) {
-      size_t offset = 0;
-      for (auto &cell : cells) {
-        _functor->SoALoader(cell, _soa, offset);
-        offset += cell.size();
+      // First resize the SoA to the required number of elements to store. This avoids resizing successively the SoA in
+      // SoALoader.
+      size_t numParticles = 0;
+#ifdef AUTOPAS_OPENMP
+      // this heuristik was taken from CellBasedParticleContainer::size()
+      const int numThreads = std::clamp(static_cast<int>(cells.size() / 100000), 1, omp_get_max_threads());
+#pragma omp parallel for num_threads(numThreads) reduction(+ : numParticles)
+#endif
+      for (size_t i = 0; i < cells.size(); ++i) {
+        numParticles += cells[i].size();
+      }
+      _soa.resizeArrays(numParticles);
+
+#ifdef AUTOPAS_OPENMP
+#pragma omp parallel for
+#endif
+      for (size_t i = 0; i < cells.size(); ++i) {
+        const auto offset = std::accumulate(cells.begin(), cells.begin() + i, 0,
+                                            [](const auto &acc, const auto &cell) { return acc + cell.size(); });
+        _functor->SoALoader(cells[i], _soa, offset, /*skipSoAResize*/ true);
```

Comment:
I actually used this heuristic for the second loop at the beginning, but the performance results were much worse than in the measurements above. Sometimes even slower than with the current version. That's why I removed it again. Either we find another heuristic, or we leave it as it is.
Accordingly, it probably makes no sense to merge the two parallel regions. It would definitely be possible to merge them with the same heuristic. After `parallel for` is an implicit barrier (I remember that pretty well from the Parallel Programming lecture). Instead of the `omp single`, we could also use an `omp master` (or I think it is now called primary) to ensure that the result of the reduction is available.

Or is there a way to have a parallel region (without `num_threads`) and then define the number of threads for each for-loop within the parallel region separately and all other threads are then idle?

Reply:
No, I don't think you can change the number of threads within a region.

Does the numThreads heuristic even help you for the first loop? Or parallelization of it in general?

Reply:
I did some measurements on my Xeon E5-2650 with the example from above that creates 8000 cells:

The `numthreads` is used in the parallel for loop that sums up the sizes. I used a timer, that measures only the time spent in the particle summation. 

| numthreads | avg time sum over all cell sizes (sec) |
|------------|----------------------------------------|
| 1          | 0.07140                                |
| 2          | 0.07286                                |
| 4          | 0.07294                                |
| 8          | 0.07160                                |

I also used an adjusted scenario that creates 205379 cells, so the heuristic will increase the number of threads:

| numthreads | avg time sum over all cell sizes (sec) |
|------------|----------------------------------------|
| 1          | 0.4541                                 |
| 2          | 0.4279                                 |

So I think this parallelization for the summation of particle buffer sizes is not really beneficial.

Reply:
But it also seems like it adds zero additional time, which means the overhead and gains cancel each other **exactly**? Looks a bit fishy to me :raised_eyebrow: But if you are sure the measurements are fine adapt the code accordingly :shrug:

Reply:
Yes, that's what I thought too. I'll check it again to be sure.

Reply:
There was actually a problem with the measurement. I accidentally measured the `resize()` as well along with the size summation. New data is in the above comment ↑

Reply:
Ok then remove it :)

Reply:
done in [454d438](https://github.com/AutoPas/AutoPas/pull/798/commits/454d4385d8cade46d7a1a712fe41c83cb622e74d)